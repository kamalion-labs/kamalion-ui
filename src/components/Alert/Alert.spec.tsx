import React from 'react';
import renderer from 'react-test-renderer';
import { Alert } from '.';

describe('Alert', () => {
  it("should return null if there's no children", () => {
    const alert = renderer.create(<Alert type="success">{null}</Alert>).toJSON();

    expect(alert).toEqual(null);
  });

  it('should render danger alert', () => {
    const alert = renderer.create(<Alert type="danger">Danger</Alert>).toJSON();

    expect(alert).toMatchSnapshot();
  });

  it('should render warning alert', () => {
    const alert = renderer.create(<Alert type="warning">Warning</Alert>).toJSON();

    expect(alert).toMatchSnapshot();
  });

  it('should render info alert', () => {
    const alert = renderer.create(<Alert type="info">Info</Alert>).toJSON();

    expect(alert).toMatchSnapshot();
  });

  it('should render success alert', () => {
    const alert = renderer.create(<Alert type="success">Success</Alert>).toJSON();

    expect(alert).toMatchSnapshot();
  });
});
