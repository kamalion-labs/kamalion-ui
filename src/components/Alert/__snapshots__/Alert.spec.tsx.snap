// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render danger alert 1`] = `
<div
  className="alert alert-danger undefined"
>
  Danger
</div>
`;

exports[`Alert should render info alert 1`] = `
<div
  className="alert alert-info undefined"
>
  Info
</div>
`;

exports[`Alert should render success alert 1`] = `
<div
  className="alert alert-success undefined"
>
  Success
</div>
`;

exports[`Alert should render warning alert 1`] = `
<div
  className="alert alert-warn undefined"
>
  Warning
</div>
`;
